[wongyx] iP

wongyx opened the pull request:
### Duke

Duke helps you to keep track of upcoming tasks that you might have. It's

- text-based
- ~~easy~~  _VERY_ EASY to use

You just need to 

1. Download it from [here](https://github.com/wongyx/ip/releases/tag/A-Jar)
2. Run the ip.jar file
3. Add some task in
4. Let Duke keep track of all your task easily

And it is **FREE**

Features include:

- [x] Managing tasks
- [ ] Reminders (coming soon) 🚧 

Here's the `main` method of Duke:

```java
public static void main(String[] args) {
    new Duke("data/tasks.txt").run();
}
```

Commit:
Add Gradle support

wongyx committed:
Added Level-1

wongyx committed:
Added Level-2

wongyx committed:
Added Level-3

wongyx committed:
Added Level-4

wongyx committed:
Added A-TextUiTesting

wongyx committed:
Added Level-5

wongyx committed:
Added Level-6

wongyx committed:
Add Level-7

wongyx committed:
Add Level-8

wongyx committed:
Merge Level-7 and Level-8

wongyx committed:
Delete data file

This file contains backend data for the program

Files containing backend data should not be on the repo, hence it is deleted

wongyx committed:
Add multiple classes and package

Most of the code is written under main.

Extracting closely related code into classes and placing them
into packages improves the code quality.

Let's create multiple classes to deal with related parts of the
code and package them together.

wongyx committed:
Merge branch 'master' of https://github.com/wongyx/ip

wongyx committed:
Merge branch 'add-gradle-support'

wongyx committed:
Add gradle support for running Duke

wongyx committed:
Add JUnit test for some classes

wongyx committed:
Add javadocs to most classes

wongyx committed:
Add feature for finding tasks

wongyx committed:
Merge branch 'branch-Level-9'

wongyx committed:
Change code to comply with coding standards

wongyx committed:
Merge branch 'branch-A-CodingStandard'

wongyx committed:
Merge JavaDoc, CodingStandard and Level-9

wongyx committed:
Fix an issue where data is not properly saved

Mark, Unmark and Delete Commands do not save after execution.

This commits fixes these issues.

wongyx committed:
Add GUI to Duke

glozxi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,149 @@
+        throws TaskNoDescriptionException, NotEnoughArgumentsException, DateTimeParseException {
+        String info = fullCommand.substring(8).trim();
+        if (info.isEmpty()) {
+            throw(new TaskNoDescriptionException("OOPS!!! The description of a deadline cannot be empty."));
```
I have noticed that you put parentheses around new thrown exceptions. I don't think that is against the coding standards, but perhaps leave a space between `throw` and `(`. This is because `throw` is a Java reserved word, so should be followed by a white space.

glozxi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,149 @@
+    public static Command parse(String fullCommand) {
+        String[] commandParts = fullCommand.split(" ", 2);
+        String commandHeader = commandParts[0];
+        switch (commandHeader) {
```
You might want to unindent switch cases.

glozxi replied:
Perhaps you could think about using enums for the command types?

pkwangjoo replied:
Perhaps you could unindent switch cases?

glozxi commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,28 @@
+package duke.commands;
+
+/** This is a command to add a task to Duke
```
Can put `/**` on separate line

glozxi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,149 @@
+import java.time.format.DateTimeParseException;
+
+/**
+ * This is a class responsible for parsing the inputs given by user
```
Perhaps put punctuation behind each parameter description. I have noticed this in other places too.

glozxi commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,91 @@
+    public Event(String description, LocalDate from, LocalDate to, boolean isDone) {
+        super(description, isDone);
+        this.from = from;
+        this.to= to;
```
Maybe a space around the `=`?

glozxi commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,20 @@
+    public abstract boolean isExit();
+
+    public abstract String execute(TaskList taskList, Ui ui, Storage storage);
+}
```
I like how you used OOP for the commands.

glozxi reviewed (commented):
Maybe change code to follow the coding standard, but overall I like it.

pkwangjoo commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,24 @@
+        Command c = Parser.parse("deadline grocery /by 19/10/2023");
+        LocalDate date = LocalDate.parse("19/10/2023", DateTimeFormatter.ofPattern("d/MM/yyyy"));
+        AddDeadlineCommand test = new AddDeadlineCommand("grocery", date);
+        //AddTodoCommand test = new AddTodoCommand("grocery");
```
Perhaps unused code can be deleted

pkwangjoo commented on src/main/java/duke/tasks/Todo.java:
```diff
@@ -0,0 +1,67 @@
+        }
+
+        Todo t = (Todo) o;
+        if (this.description.equals(t.description)) {
```
A minor nit but perhaps this boolean value could be returned directly.

pkwangjoo commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+        this.counter = tasks.size();
```
Maybe you could name `counter` as `size` for clearer naming?

wongyx committed:
Improve code quality

Code has some parts that does not follow coding standards.

Changing the code to follow coding standards will help improve
readability and quality of the code.

wongyx committed:
Add assertions

wongyx committed:
Merge pull request #2 from wongyx/branch-A-CodeQuality

Improve code quality

wongyx committed:
Merge branch 'master' into branch-A-Assertions

wongyx committed:
Merge pull request #1 from wongyx/branch-A-Assertions

Add assertions

wongyx committed:
Add feature to reject duplicated tasks

wongyx committed:
Fix a bug where user can mark invalid task

wongyx committed:
Update user guide and Ui.png

wongyx committed:
Release v0.2
